gammapy/spectrum/make.py

```diff
@@ -195,3 +195,120 @@ def run(self, observation, selection=None):
             kwargs["edisp"] = self.make_edisp(observation)
 
         return SpectrumDataset(**kwargs)
+
+
+class SafeMaskMaker:
+    """Make safe data range mask for a given observation.
+
+    Parameters
+    ----------
+    methods : {"aeff-default", "aeff-max", "edisp-bias"}
+        Method to use for the safe energy range. Can be a
+        list with a combination of those. Resulting masks
+        are combined with logical `and`. "aeff-default"
+        uses the energy ranged specified in the DL3 data
+        files, if available.
+    aeff_percent : float
+        Percentage of the maximal effective area to be used
+        as lower energy threshold for method "aeff-max".
+    bias_percent : float
+        Percentage of the energy bias to be used as lower
+        energy threshold for method "edisp-bias"
+    """
+
+    def __init__(
+        self, methods=["aeff-default"], aeff_percent=10, bias_percent=10
+    ):
+        self.methods = methods
+        self.aeff_percent = aeff_percent
+        self.bias_percent = bias_percent
+
+    @staticmethod
+    def make_mask_energy_aeff_default(dataset, observation):
+        """Make safe energy mask from aeff default.
+
+        Parameters
+        ----------
+        dataset : `Dataset`
+            Dataset to compute mask for.
+        observation: `DataStoreObservation`
+            Observation to compute mask for.
+
+        Returns
+        -------
+        mask_safe : `~numpy.ndarray`
+            Safe data range mask.
+        """
+        try:
+            e_max = observation.aeff.high_threshold
+            e_min = observation.aeff.low_threshold
+        except KeyError:
+            log.warning(f"No thresholds defined for obs {observation}")
+            e_min, e_max = None, None
+
+        mask_safe = dataset.counts.energy_mask(emin=e_min, emax=e_max)
```

Comment:
Just return `mask_safe`, no point in creating a temp variable.

Reply:
Done